src/main/groovy/edu/oregonstate/mist/locations/resources/LocationResource.groovy

```diff
@@ -0,0 +1,42 @@
+package edu.oregonstate.mist.locations.resources
+
+import com.codahale.metrics.annotation.Timed
+import com.google.common.base.Optional
+import edu.oregonstate.mist.locations.core.Location
+import edu.oregonstate.mist.locations.db.LocationDAO
+import io.dropwizard.jersey.params.LongParam
+
+import javax.ws.rs.Consumes
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path("/locations")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class LocationResource {
+    private final LocationDAO locationDAO
+
+    //@todo: is this really needed?
+    LocationResource(LocationDAO locationDAO) {
+        this.locationDAO = locationDAO
+    }
+
+    @GET
+    @Timed
+    @Path('{id: \\d+}')
+    Location getById(@PathParam('id') LongParam id) {
+        final Location location = locationDAO.getLocationById(id.get())
+
+        if (!location) {
+            //@todo - does this also set the correct headers?
```

Comment:
I did take a look at your code. I checked and both of them do the same. The WebApplicationException sets the status code header correctly. It just relies on jetty to display the default error page for the status code.
It probably would be beneficial to have a class in the skeleton app that handles the custom error messages that we want. This is more than just a simple error message. We want to specify error messages in json as outlined in the best practices wiki page.